swiftui/ChartboostMediationDemo/Ads/Interstitial/InterstitialAdController.swift

```diff
@@ -44,47 +50,105 @@ class InterstitialAdController: NSObject, ObservableObject {
             return
         }
 
-        // Create an interstitial ad from the Chartboost Mediation ad provider.
-        guard let interstitialAd = chartboostMediation.interstitialAdProvider(with: self, andPlacementName: placementName) else {
-            // It could fail if the placement name is invalid.
-            print("[Error] failed to create interstitial advertisement for placement '\(placementName)'")
-            return
-        }
-        self.interstitialAd = interstitialAd
-
         // Notify the demo UI
         activityState = .running
 
-        // Associate any provided keywords with the advertisement before it is loaded.
-        interstitialAd.keywords = keywords
+        if fullscreenAPI {
+            // loadFullscreenAd expects keywords to be `[String : String]` instead of `HeliumKeywords?`.
+            let keywords = keywords?.dictionary ?? [:]
+            let request = ChartboostMediationAdLoadRequest(placement: placementName, keywords: keywords)
+            // Load the fullscreen ad, which will make a request to the network. Upon completion, a
+            // ChartboostMediationFullscreenAdLoadResult will be passed to the completion block.
+            chartboostMediation.loadFullscreenAd(with: request) { [weak self] result in
```

Comment:
This `[weak self]` is only here because I saw [that's how it's done in Canary](https://github.com/ChartBoost/ios-helium-sdk/blob/fa270e05fb4fe8ac33b1c183ee6d98e3cf8158f0/HeliumCanary/HeliumCanary/Placements/Types/Fullscreen/FullscreenPlacementViewController.swift#L158).

In the case of `ChartboostMediationFullscreenAd.show(...)` I think I see why it's necessary: because `self` holds a reference to the ad, which holds a reference to the closure, so the closure holding a strong reference to `self` would cause a cycle.

But `chartboostMediation.loadFullscreenAd(...)` is a call to the SDK, so I don't think that leads to a cycle. Is `weak self` necessary because `chartboostMediation` will outlive this ad controller and it might simply hold onto that reference and prevent this object from ever deallocating?

Reply:
Your last statement is correct and why the `[weak self]` is necessary here. Otherwise the controller will be kept artificially alive until the load finishes, which may be long after the user has moved on from this screen.